pkg/assembler/nodes.go

```diff
@@ -318,3 +354,46 @@ func (e ContainsEdge) PropertyNames() []string {
 func (e ContainsEdge) IdentifiablePropertyNames() []string {
 	return []string{}
 }
+
+// MetadataFor is an edge that represents the fact that an
+// a metadata node represents metadata for an `ArtifactNode/PackageNode`
+// Only one of each side of the edge should be defined.
+type MetadataForEdge struct {
+	// From node
+	MetadataNode MetadataNode
+	// To node
+	ForArtifact ArtifactNode
+	ForPackage  PackageNode
```

Comment:
I'm not sure we can have an edge with 3 nodes. Looking at Nodes() below it looks like we either have an artifact or a package, but never both? Should we document this? (more of a nit)

Reply:
hmm - i added a comment in the struct definition - do you think we need to be more explicit?

Reply:
I think we should be good